pkg/odo/cli/dev/dev.go

```diff
@@ -230,8 +234,16 @@ func (o *DevOptions) Run(ctx context.Context) error {
 	scontext.SetProjectType(ctx, devFileObj.Data.GetMetadata().ProjectType)
 	scontext.SetDevfileName(ctx, devFileObj.GetMetadataName())
 
-	d := Handler{}
-	err = o.clientset.DevClient.Watch(devFileObj, path, o.ignorePaths, o.out, &d, o.ctx)
+	if o.noWatchFlag {
+		log.Finfof(log.GetStdout(), "\n"+watch.CtrlCMessage)
+		for {
```

Comment:
I'm not an expert with channels and goroutines, but I feel like attempting to read from the Done channel with `<-o.ctx.Done()` is already a blocking call, which won't continue until the `o.ctx` is done.
So I think this infinite `for` loop is not necessary, especially since there is no other `select` cases there, no?
I quickly tested without the `for` loop and it worked as expected. What do you think?

Reply:
That's right. I had initially created a `select` with a single case of listening on `o.ctx`. I removed the `select`, but didn't realise that `for` could be removed as well. Thanks for pointing. 👍🏾

Reply:
Done. Thanks for the catch.